kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ConsumerOffsetsDaoImpl.java

```diff
@@ -0,0 +1,165 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ConsumerGroupDescription;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsOptions;
+import org.apache.kafka.clients.admin.ListOffsetsOptions;
+import org.apache.kafka.clients.admin.ListOffsetsResult.ListOffsetsResultInfo;
+import org.apache.kafka.clients.admin.MemberDescription;
+import org.apache.kafka.clients.admin.OffsetSpec;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.ConsumerGroupState;
+import org.apache.kafka.common.IsolationLevel;
+import org.apache.kafka.common.TopicPartition;
+
+final class ConsumerOffsetsDaoImpl implements ConsumerOffsetsDao {
+
+  private final Admin kafkaAdminClient;
+  @Inject
+  ConsumerOffsetsDaoImpl(
+      Admin kafkaAdminClient
+  ) {
+    this.kafkaAdminClient = kafkaAdminClient;
+  }
+
+  @AutoValue
+  abstract static class MemberId {
+
+    abstract String getConsumerId();
+
+    abstract String getClientId();
+
+    abstract Optional<String> getInstanceId();
+
+    public static Builder builder() {
+      return new AutoValue_ConsumerOffsetsDaoImpl_MemberId.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+
+      abstract Builder setConsumerId(String consumerId);
+
+      abstract Builder setClientId(String clientId);
+
+      abstract Builder setInstanceId(Optional<String> instanceId);
+
+      abstract MemberId build();
+    }
+  }
+
+  @Override
+  public CompletableFuture<Map<TopicPartition, MemberId>> getMemberIds(
+      ConsumerGroupDescription cgDesc
+  ) {
+    Map<TopicPartition, MemberId> tpMemberIds = new HashMap<>();
+    for (MemberDescription memberDesc : cgDesc.members()) {
+      for (TopicPartition tp : memberDesc.assignment().topicPartitions()) {
+        MemberId memberId =
+            MemberId.builder()
+                .setConsumerId(memberDesc.consumerId())
+                .setClientId(memberDesc.clientId())
+                .setInstanceId(memberDesc.groupInstanceId())
+                .build();
+        tpMemberIds.put(tp, memberId);
+      }
+    }
+    return CompletableFuture.completedFuture(tpMemberIds);
+  }
+
+  @Override
+  public CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> getCurrentOffsets(
+      String consumerGroupId
+  ) {
+    return KafkaFutures.toCompletableFuture(
+        kafkaAdminClient.listConsumerGroupOffsets(
+            consumerGroupId, new ListConsumerGroupOffsetsOptions())
+            .partitionsToOffsetAndMetadata());
+  }
+
+  @Override
+  public CompletableFuture<Map<TopicPartition, ListOffsetsResultInfo>> getLatestOffsets(
+      IsolationLevel isolationLevel,
```

Comment:
You use the same isolation level in both ConsumerGroupLagManagerImpl and ConsumerLagManagerImpl (READ_COMMITED). Why even bother receiving this as a parameter? Why don't you define it here (in this class) instead, and remove the parameter?